### PR TITLE
Regenerated bindings from 0.34.x

### PR DESCRIPTION
**Pull request by et1975**
Beaware: breaking changes.

Took TS definitions from DT, reimported and reorganized to make it compile.
Regenerated Helpers DSL and added unifying types to avoid copy-pasting props across DUs.
Moved less used DUs into enclosing modules to avoid them hiding commonly used DU cases.

### CODE REVIEW

File: react-native/Fable.Helpers.ReactNative.fs [hunk 1]
```diff
@@ -157,722 +359,1335 @@ module Props =
...
         | BorderRightColor of string
+        | BorderRightWidth of float
+        | BorderStyle of (* TODO StringEnum solid | dotted | dashed *) string
         | BorderTopColor of string
         | BorderTopLeftRadius of float
         | BorderTopRightRadius of float
+        | BorderTopWidth of float
         | Opacity of float
-        | Overflow of string
+        | Overflow of (* TODO StringEnum visible | hidden *) string
```
Comment (alfonsogarciacaro) [inline react-native/Fable.Helpers.ReactNative.fs]: Shall we add these `StringEnum`s (also for BorderStyle) above by hand? Unfortunately, there are many of these all over the place and could be a bit tedious.

File: react-native/Fable.Helpers.ReactNative.fs [hunk 2]
```diff
@@ -157,722 +359,1335 @@ module Props =
...
+        | OnDateChange of Func<DateTime, unit>
+        | TimeZoneOffsetInMinutes of float
+        | Ref of Ref<DatePickerIOS>
+        interface IDatePickerIOSProperties
+
+    [<KeyValueList>]
+    type DrawerLayoutAndroidProperties =
+        | DrawerBackgroundColor of string
+        | DrawerLockMode of DrawerLockMode 
+        | DrawerPosition of obj
```
Comment (alfonsogarciacaro) [inline react-native/Fable.Helpers.ReactNative.fs]: Unfortunately, this is overwriting [my change from yesterday](https://github.com/fable-compiler/fable-react/commit/0ad6306f4e046f426ff4b639f3c20ab7fdf39381), could we recover that?

File: react-native/Fable.Helpers.ReactNative.fs [hunk 3]
```diff
@@ -157,722 +359,1335 @@ module Props =
...
-        | OnShouldStartLoadWithRequest of (unit -> bool)
-        | RenderError of (unit -> ViewStatic)
-        | RenderLoading of (unit -> ViewStatic)
-        | ScrollEnabled of bool
-        | StartInLoadingState of bool
-        | Style of ViewStyle
-        | Url of string
-        interface IWebViewProperties
+    type LayoutAnimationConfig =
+        | Delay of float //REQUIRED!
```
Comment (alfonsogarciacaro) [inline react-native/Fable.Helpers.ReactNative.fs]: These `REQUIRED` comments could go on top of the case with triple slash so they're visible in the autocompletion tool of the IDE.

```fsharp
/// REQUIRED!
| Delay of float
```

**Comment by et1975** [comment]
Dealing with breaking changes w.r.t Styles/Flex/etc:
 The field, constructor or member 'Margin' is not defined (L37,18-L37,24) (Styles.fs)
> The field, constructor or member 'Margin' is not defined (L44,18-L44,24) (Styles.fs)
> The field, constructor or member 'Margin' is not defined (L52,18-L52,24) (Styles.fs)
> The field, constructor or member 'AlignSelf' is not defined (L58,18-L58,27) (Styles.fs)
> The field, constructor or member 'Padding' is not defined (L59,18-L59,25) (Styles.fs)
> The field, constructor or member 'JustifyContent' is not defined (L63,18-L63,32) (Styles.fs)
> The field, constructor or member 'Flex' is not defined (L64,18-L64,22) (Styles.fs)

Just remove DU type name, and it should find the right source for all the cases.

File: react-native/Fable.Helpers.ReactNative.fs (hunk 3 above)
Comment (et1975) [inline react-native/Fable.Helpers.ReactNative.fs]: That was more of a comment for my future self - I'm not convinced there's actually such a thing when it comes to react props. But I like your idea.

File: react-native/Fable.Helpers.ReactNative.fs (hunk 2 above)
Comment (et1975) [inline react-native/Fable.Helpers.ReactNative.fs]: Sorry, the TS definitions I pulled didn't have some of those and somehow I missed it reconciling the differences. Amended the commit.

File: react-native/Fable.Helpers.ReactNative.fs (hunk 1 above)
Comment (et1975) [inline react-native/Fable.Helpers.ReactNative.fs]: Amended the commit, now with 100% more enums!

File: react-native/Fable.Helpers.ReactNative.fs [hunk 4]
```diff
@@ -108,11 +129,320 @@ module Props =
...
+    type TextDecorationStyle = 
+        | Solid | Double | Dotted | Dashed
+
+    [<StringEnum; RequireQualifiedAccess>]
+    type FontStyle = 
+        | Normal | Italic 
+        
+    [<StringEnum; RequireQualifiedAccess>]
+    type WritingDirection =
+        | Auto | Ltr | Rtl
```
Comment (forki) [inline react-native/Fable.Helpers.ReactNative.fs]: can we use full names here and just put Ltr in the generated JS?

File: react-native/Fable.Helpers.ReactNative.fs [hunk 5]
```diff
@@ -108,11 +129,320 @@ module Props =
...
+        | [<CompiledName("500")>] F500 
+        | [<CompiledName("600")>] F600 
+        | [<CompiledName("700")>] F700 
+        | [<CompiledName("800")>] F800 
+        | [<CompiledName("900")>] F900
+
+    [<StringEnum; RequireQualifiedAccess>]
+    type TextDecorationLine =
+        | None | Underline 
+        | ``Line-through``
```
Comment (forki) [inline react-native/Fable.Helpers.ReactNative.fs]: Please use real F# names + CompiledName attribute

File: react-native/Fable.Helpers.ReactNative.fs [hunk 6]
```diff
@@ -137,12 +463,16 @@ module Props =
         | BorderTopWidth of float
         | BorderWidth of float
         | Bottom of float
-        | Flex of int
-        | FlexDirection of string
-        | FlexWrap of string
+        | Flex of float
```
Comment (forki) [inline react-native/Fable.Helpers.ReactNative.fs]: can this be any number?

File: react-native/Fable.Helpers.ReactNative.fs [hunk 7]
```diff
@@ -157,722 +487,1334 @@ module Props =
...
+        | ZIndex of float
         interface IFlexStyle
 
-    [<KeyValueList>]
-    type IViewStyle =
-        interface end
 
     [<KeyValueList>]
     type ViewStyle =
-        // FlexStyle
```
Comment (forki) [inline react-native/Fable.Helpers.ReactNative.fs]: why did you remove this?

File: react-native/Fable.Helpers.ReactNative.fs (hunk 3 above)
Comment (forki) [inline react-native/Fable.Helpers.ReactNative.fs]: I tried to put required props in the ctor calls.

File: react-native/Fable.Helpers.ReactNative.fs (hunk 6 above)
Comment (et1975) [inline react-native/Fable.Helpers.ReactNative.fs]: Not sure, but looking at https://github.com/facebook/css-layout/tree/master/csharp/Facebook.CSSLayout the values are floats, which is consistent with other values, like margin.

File: react-native/Fable.Helpers.ReactNative.fs (hunk 4 above)
Comment (et1975) [inline react-native/Fable.Helpers.ReactNative.fs]: Personally, I'd want to stick to the original RN names because searching for usage help might be easier that way. In this specific case `Rtl` and `Ltr` are common UI design abbreviations.
